File: src/MixNetFx/FacInterfaces/ImportFormsAndHelper/frmTimeLogResultTable.vb
Imports System.Windows.Forms
Imports Microsoft.Reporting.WinForms

Public Class frmTimeLogResultTable

    Private myTimeLogData As TimeDataTable
    Private myProductionDate As Date

    Private Sub frmTimeLogResultTable_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Dim tmpParams As New List(Of ReportParameter)
        tmpParams.Add(New ReportParameter("ReportDate", myProductionDate.ToString))
        Me.rvEmployeeTimeLogResult.LocalReport.SetParameters(tmpParams)
        Me.rvWorksiteTimeLogResult.LocalReport.SetParameters(tmpParams)
        Me.TimeDataRowBindingSource.DataSource = myTimeLogData
        Me.rvEmployeeTimeLogResult.RefreshReport()
        Me.rvWorksiteTimeLogResult.RefreshReport()
    End Sub

    Overloads Function ShowDialog(ByVal timeLogData As TimeDataTable, ByVal ProductionDate As Date) As DialogResult
        myProductionDate = ProductionDate
        myTimeLogData = timeLogData
        Return MyBase.ShowDialog
    End Function
End Class

File: src/NetFxCSharp/Facesso.Interfaces/ImportFormsAndHelper/frmTimeLogResultTable.cs
using Microsoft.Reporting.WinForms;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Xml.Linq;

namespace Facesso.Interfaces
{
    public partial class frmTimeLogResultTable
    {
        private TimeDataTable myTimeLogData;
        private System.DateTime myProductionDate;
        private void frmTimeLogResultTable_Load(System.Object sender, System.EventArgs e)
        {
            List<ReportParameter> tmpParams = new List<ReportParameter>();
            tmpParams.Add(new ReportParameter("ReportDate", myProductionDate.ToString()));
            this.rvEmployeeTimeLogResult.LocalReport.SetParameters(tmpParams);
            this.rvWorksiteTimeLogResult.LocalReport.SetParameters(tmpParams);
            this.TimeDataRowBindingSource.DataSource = myTimeLogData;
            this.rvEmployeeTimeLogResult.RefreshReport();
            this.rvWorksiteTimeLogResult.RefreshReport();
        }

        public DialogResult ShowDialog(TimeDataTable timeLogData, System.DateTime ProductionDate)
        {
            myProductionDate = ProductionDate;
            myTimeLogData = timeLogData;
            return base.ShowDialog();
        }

        public frmTimeLogResultTable()
        {
            this.Load += frmTimeLogResultTable_Load;
            InitializeComponent();
        }
    }
}